lib/service/service.go

```diff
@@ -3268,6 +3269,12 @@ func (process *TeleportProcess) setupProxyTLSConfig(conn *Connector, tsrv revers
 		}
 		tlsConfig = m.TLSConfig()
 		utils.SetupTLSConfig(tlsConfig, cfg.CipherSuites)
+
+		// If ACME protocol was enabled add all known TLS Routing protocols.
+		// Go 1.17 introduced strict ALPN https://golang.org/doc/go1.17#ALPN If a client protocol is not recognized
+		// the TLS handshake will fail.
+		supportedProtocols := alpncommon.ProtocolsToString(alpncommon.SupportedProtocols)
+		tlsConfig.NextProtos = apiutils.Deduplicate(append(tlsConfig.NextProtos, supportedProtocols...))
```

Comment:
Should this also be applied to `tlsConfig` if ACME isn't enabled (i.e. outside this "if..else" block)?

Reply:
In current behavior this is not necessary. For no ACME flow the Server TLS.NextProtos list is empty thus it either client or server supported protocols list is empty the ALPN  strict behavior will allow for the handshake.